subsys/fs/Kconfig.littlefs

```diff
@@ -8,6 +8,7 @@ config FILE_SYSTEM_LITTLEFS
 	depends on FILE_SYSTEM
 	depends on FLASH_MAP
 	depends on FLASH_PAGE_LAYOUT
+	depends on FLASH_MAP_DEV_NAME_IN_FLASH_AREA
```

Comment:
Only reason of forcing device_name is the line subsys/fs/littlefs_fs.c 638:
``` C
 LOG_ERR("can't get flash device: %s", log_strdup(fs->area->fa_dev_name));
```

I would replace it by:
``` C
LOG_ERR("can't get flash device of flash area id: %d", area_id);
```
and resign from this dependency.

Reply:
+ modify a log line in `samples/subsys/fs/littlefs/src/main.c`

Reply:
I left it there because this can be decided afterwards and wanted the minimal change in code outside of the flash API.